internal/monitors/replicationstatus.go

```diff
@@ -0,0 +1,63 @@
+package monitors
+
+import (
+	"log"
+	"time"
+
+	"cloudant.com/cloudant_exporter/internal/utils"
+	"github.com/IBM/cloudant-go-sdk/cloudantv1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ReplicationStatusMonitor struct {
+	Cldt     *cloudantv1.CloudantV1
+	Interval time.Duration
+	FailBox  *utils.FailBox
+}
+
+var (
+	replicatonStatus = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "cloudant_replication_status",
+			Help: "Current status per replication",
+		},
+		[]string{"docid", "source", "target", "status"},
+	)
+)
+
+func (rc *ReplicationStatusMonitor) Name() string {
+	return "ReplicationStatusMonitor"
+}
+
+func (rc *ReplicationStatusMonitor) Retrieve() error {
+	var skip int = 0
+	var batchSize int = 100
+
+	// repeat until we get a smaller batch than we asked for
+	for {
+		// fetch scheduler jobs
+		getSchedulerJobsOptions := rc.Cldt.NewGetSchedulerJobsOptions()
+		getSchedulerJobsOptions.SetLimit(int64(batchSize))
+		getSchedulerJobsOptions.SetSkip(int64(skip))
```

Comment:
Should this call `_scheduler/docs` rather than `_scheduler/jobs`? Do things disappear from jobs after a while? I'm not sure of the behaviour here, whether it's what we want or it's not 😕

Reply:
I think switching to `../docs` is better, having read the dococumentation.